### PR TITLE
Add JOB benchmark dataset [1/N]  (imdb dataset)

### PR DESCRIPTION
**Pull request by doupache**
## Which issue does this PR close?
Partial Closes https://github.com/apache/datafusion/issues/12311

```sh
cd benchmarks/   
./bench.sh data imdb
```
All IMDB tables are now generated in `benchmarks/data/imdb/*.parquet`

## Rationale for this change
Add imdb dataset for the JOB benchmarking

## What changes are included in this PR?
Download the the dataset and convert it to parquet files.

## Are these changes tested?
Just like running to generate tpch dataset:
`./bench.sh data tpch`

run:
`./bench.sh data imdb`

## Are there any user-facing changes?
no

### CODE REVIEW

**Comment by doupache** [comment]
Unlike TPCH, which uses table-named folders with partitioned parquet files, IMDB has smaller tables (largest is 360MB). 
We can convert each IMDB table to a single, non-partitioned parquet file.

image

**Comment by austin362667** [comment]
Thanks @doupache for paving the way! Got few nit suggestion.
1. Do we prefer benchmark name as `imdb` or `job`?
2. Cloud you add `[1/N]` in the beginning of the PR title to help us track the follow-ups progress?

**Review by austin362667 — APPROVED** [review]

**Comment by doupache** [comment]
Thanks @austin362667 for the suggestions. **IMDB** is more suitable than **JOB** as it's specific and avoids confusion. **Job** can be used in many different contexts.

Adding 'progress' to the title is also a good idea 👍

File: benchmarks/src/imdb/convert.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            let config = SessionConfig::new().with_batch_size(self.batch_size);
+            let ctx = SessionContext::new_with_config(config);
+
+            let mut csv = ctx.read_csv(&input_path, options).await?;
+
+            // Select all apart from the padding column
+            let selection = csv
+                .schema()
+                .iter()
+                .take(schema.fields.len() - 1)
```
Comment (austin362667) [inline benchmarks/src/imdb/convert.rs:77]: Why takes all columns except the last one?

```suggestion
                .take(schema.fields.len())
```

**Review by austin362667 — CHANGES_REQUESTED** [review]

**Comment by alamb** [comment]
Thanks @doupache  -- I started the CI jobs, and I will try and test this out manually locally over the next few days

**Comment by doupache** [comment]
Thanks @austin362667  and @alamb. 

I have updated the PR and learned some Cargo tips from @austin362667. 
Using debug build during development is much faster.


```sh 
#1
cd benchmarks && cargo build 

#2 
cargo run --bin imdb -- convert --input ./data/imdb/ --output ./data/imdb/ --format parquet
```


i also test all 21 parquet  like follwoing.

```sql 
# create table 
CREATE EXTERNAL TABLE name (
    id INTEGER NOT NULL PRIMARY KEY,
    name STRING NOT NULL,
    imdb_index STRING,
    imdb_id INTEGER,
    gender STRING,
    name_pcode_cf STRING,
    name_pcode_nf STRING,
    surname_pcode STRING,
    md5sum STRING
)
STORED AS PARQUET
LOCATION '../benchmarks/data/imdb/temp/name.parquet';

# read 
SELECT * FROM name LIMIT 5;
```

**Comment by austin362667** [comment]
> Unlike TPCH, which uses table-named folders with partitioned parquet files, IMDB has smaller tables (largest is 360MB).
We can convert each IMDB table to a single, non-partitioned parquet file.

Sure! I think maybe it's because historical reasons that `ParquetExec` didn't support parallel execution for single Parquet file back then. Now it's being supported in https://github.com/apache/datafusion/pull/5057

**Review by austin362667 — APPROVED** [review]
Thanks @doupache for the contribution and @alamb for the review~
I'll 1. add `imdb` help text. and fix what @andygrove told in PR [2/N] https://github.com/apache/datafusion/pull/1252,
2. non-neg id use UInt32
3 use single context.

**Comment by alamb** [comment]
Let's keep improving things in the next PR. Thanks @austin362667

**Pull request merged**